src/runtime/classmanager.cs

```diff
@@ -180,6 +253,7 @@ private static void InitClassBase(Type type, ClassBase impl)
                         // TODO: deprecate __overloads__ soon...
                         Runtime.PyDict_SetItemString(dict, "__overloads__", ctors.pyHandle);
                         Runtime.PyDict_SetItemString(dict, "Overloads", ctors.pyHandle);
+                        ctors.DecrRefCount();
```

Comment:
I agree. This also ties in the other comment about using a .NET or Python weakref. It's never clear to me in which cases any class of the ManagedType owns the reference or not (or even if they should own it). This PR is getting really big, so I think it would be better that it goes in a separate PR.

Reply:
It's just a bug fix. I didn't change the behavior of they own it or not. Behaviors changes like these are not the parts of the shutdown feature.